virtual_rainforest/models/animals/animal_communities.py

```diff
@@ -163,7 +177,12 @@ def forage_community(self) -> None:
 
 
         """
-        plant_list = [self.plant_community]
+        # Generate the plant resources for foraging.
+        plant_community: PlantResources = PlantResources(
+            data=self.data, cell_id=self.community_key
+        )
+
```

Comment:
Just making sure that I understand what is going on here - this is generating a new `PlantResource` object every time `forage_community` is called so that it can be initialized with the current state of data? 

Is this a workaround so that feedbacks can be avoided for now?

On runs, should I then expect animals to treat plants as unlimited until such a time where they are completely consumed each tick?

Reply:
That's what it is doing and it is not a workaround. The `plants` model is not accessible in an object oriented way to the `animals` model, so the `PlantResources` will always be an interface to the data. The `PlantResources` just exposes the current data in a way that is compatible with your protocols for `animals` model. What will need to happen is that `PlantResources` writes back data for herbivory, that the `plants` model can then integrate. 

There is a timing issue there, but we can probably estimate GPP as some kind of average of the original leaf mass and the remaining leaf mass. That's Milestone 2 though.

Reply:
> On runs, should I then expect animals to treat plants as unlimited until such a time where they are completely consumed each tick?

It isn't unlimited because the energy gets consumed as with your other resources. It's just that the consumption doesn't feedback to `plants` so that boatload of leaf mass will currently reappear magically at the next tick.